services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -245,6 +250,62 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     }
   }
 
+  @Override
+  public void grpcRead(GrpcHandlerContext ctx, GrpcHandlerPipeline pipeline) {
+    if (ctx.hasError()) {
+      pipeline.processRequest(ctx);
+      return;
+    }
+
+    if (ctx.getGrpcStatsContext() == null) {
+      // new request
+      GrpcStatsContext statsContext = new GrpcStatsContext(singleGetStats, multiGetStats, computeStats);
+
+      ctx.setGrpcStatsContext(statsContext);
+      pipeline.processRequest(ctx);
+    } else {
+      // existing request
```

Comment:
Does this mean it's a multi-part request? In the old implementation, we track the request part count, as we have found it to have a big influence on performance. Maybe this cannot happen in gRPC, but I suppose it probably can, so it may be good to track it as well.

Reply:
As of right now, we are only sending Unary requests to client/server. In this case, the entire request will be in one part. When message streaming is enabled for the gRPC service, we might want to keep track of stats for each message that is sent. For now, I will remove the second part -- it will never be invoked in this case.